<a href="https://colab.research.google.com/github/hbabar1215/slp_project_example/blob/main/preliminary_slp_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [102]:
import pandas as pd

In [103]:
df = pd.read_csv("slpdata_numerical.csv")
df

StartDate  \
0                                            Start Date   
1     {"ImportId":"startDate","timeZone":"America/De...   
2                                   2025-07-31 05:41:54   
3                                   2025-07-31 05:41:54   
4                                   2025-07-31 05:41:55   
...                                                 ...   
1097                                2025-07-31 17:04:12   
1098                                2025-07-31 17:04:13   
1099                                2025-07-31 17:04:13   
1100                                2025-07-31 17:04:13   
1101                                2025-07-31 17:04:13   

                                                EndDate  \
0                                              End Date   
1     {"ImportId":"endDate","timeZone":"America/Denv...   
2                                   2025-07-31 05:41:54   
3                                   2025-07-31 05:41:54   
4                                   2025-07-31 05:41:55   
...                                                 ...   
1097                                2025-07-31 17:04:12   
1098                                2025-07-31 17:04:13   
1099                                2025-07-31 17:04:13   
1100                                2025-07-31 17:04:13   
1101                                2025-07-31 17:04:13   

                     Status                 IPAddress  \
0             Response Type                IP Address   
1     {"ImportId":"status"}  {"ImportId":"ipAddress"}   
2                         2                       NaN   
3                         2                       NaN   
4                         2                       NaN   
...                     ...                       ...   
1097                      2                       NaN   
1098                      2                       NaN   
1099                      2                       NaN   
1100                      2                       NaN   
1101                      2                       NaN   

                     Progress    Duration (in seconds)  \
0                    Progress    Duration (in seconds)   
1     {"ImportId":"progress"}  {"ImportId":"duration"}   
2                         100                        0   
3                         100                        0   
4                         100                        0   
...                       ...                      ...   
1097                      100                        0   
1098                      100                        0   
1099                      100                        0   
1100                      100                        0   
1101                      100                        0   

                     Finished  \
0                    Finished   
1     {"ImportId":"finished"}   
2                           1   
3                           1   
4                           1   
...                       ...   
1097                        1   
1098                        1   
1099                        1   
1100                        1   
1101                        1   

                                           RecordedDate  \
0                                         Recorded Date   
1     {"ImportId":"recordedDate","timeZone":"America...   
2                                   2025-07-31 05:41:54   
3                                   2025-07-31 05:41:55   
4                                   2025-07-31 05:41:55   
...                                                 ...   
1097                                2025-07-31 17:04:12   
1098                                2025-07-31 17:04:13   
1099                                2025-07-31 17:04:13   
1100                                2025-07-31 17:04:13   
1101                                2025-07-31 17:04:13   

                    ResponseId                 RecipientLastName  ...  \
0                  Response ID               Recipient Last Name  .

In [104]:
# drop rows 0 and 1
df_clean = df.drop([0,1])
df_clean

StartDate              EndDate Status IPAddress Progress  \
2     2025-07-31 05:41:54  2025-07-31 05:41:54      2       NaN      100   
3     2025-07-31 05:41:54  2025-07-31 05:41:54      2       NaN      100   
4     2025-07-31 05:41:55  2025-07-31 05:41:55      2       NaN      100   
5     2025-07-31 05:41:55  2025-07-31 05:41:55      2       NaN      100   
6     2025-07-31 05:41:55  2025-07-31 05:41:55      2       NaN      100   
...                   ...                  ...    ...       ...      ...   
1097  2025-07-31 17:04:12  2025-07-31 17:04:12      2       NaN      100   
1098  2025-07-31 17:04:13  2025-07-31 17:04:13      2       NaN      100   
1099  2025-07-31 17:04:13  2025-07-31 17:04:13      2       NaN      100   
1100  2025-07-31 17:04:13  2025-07-31 17:04:13      2       NaN      100   
1101  2025-07-31 17:04:13  2025-07-31 17:04:13      2       NaN      100   

     Duration (in seconds) Finished         RecordedDate         ResponseId  \
2                        0        1  2025-07-31 05:41:54  R_6fn5ekDobUP1m5g   
3                        0        1  2025-07-31 05:41:55  R_3KR0mBnSZdjoJAa   
4                        0        1  2025-07-31 05:41:55  R_5gqVnk14OjcBmXI   
5                        0        1  2025-07-31 05:41:55  R_0HUEeMmeNFs5WZw   
6                        0        1  2025-07-31 05:41:55  R_7TYgOzdHL53Y5Ho   
...                    ...      ...                  ...                ...   
1097                     0        1  2025-07-31 17:04:12  R_8x0fqyHU55k51Eq   
1098                     0        1  2025-07-31 17:04:13  R_72OeMk6x3Bgz7AG   
1099                     0        1  2025-07-31 17:04:13  R_2rzg4pqMXolgfbg   
1100                     0        1  2025-07-31 17:04:13  R_6G9pvj6waxWi4RM   
1101                     0        1  2025-07-31 17:04:13  R_9GEi9QDgBzaApEi   

     RecipientLastName  ... Q13_1 Q13_2 Q13_3 Q13_4  \
2                  NaN  ...     1     4     5     3   
3                  NaN  ...     3     2     1     1   
4                  NaN  ...     4     3     3     4   
5                  NaN  ...     1     2     5     4   
6                  NaN  ...     3     4     3     5   
...                ...  ...   ...   ...   ...   ...   
1097               NaN  ...     3     3     1     1   
1098               NaN  ...     4     3     1     2   
1099               NaN  ...     3     2     3     2   
1100               NaN  ...     5     2     4     4   
1101               NaN  ...     3     2     4     5   

                                                    Q14  \
2     Elementum nibh nonummy orci? Nec dui tempora e...   
3     Risus lorem. Convallis convallis lectus sodale...   
4     Risus! Mattis scelerisque gravida iaculis arcu...   
5     Turpis scelerisque vestibulum risus wisi! Nec ...   
6     Rutrum! Volutpat, iaculis eu luctus. Dolorem e...   
...                                                 ...   
1097  Iaculis consectetuer! Tellus. Primis cras mole...   
1098  Pellentesque tempora! Primis lorem? Sem laoree...   
1099  Nec, commodo temporibus elementum pellentesque...   
1100  Erat est accumsan sed odio, platea porta dictu...   
1101  Consectetuer. Pellentesque? Ullamcorper platea...   

                                                    Q15  \
2     Iaculis placerat. Gravida sit. Proin primis ar...   
3     Purus magnis porta dolorem tempora. Dignissim!...   
4     Natoque cras curabitur aliquam! Urna, ipsum. I...   
5     Aliquam mauris quam facilisi curabitur. Justo ...   
6     Molestie ultrices lacus praesent vestibulum? E...   
...                                                 ...   
1097  Lacus lacus tincidunt urna vitae vel non donec...   
1098  Velit ullamcorper facilisi gravida? Facilisis ...   
1099  Est venenatis magna. Phasellus potenti sodales...   
1100  Curabitur. Sollicitudin et tortor pharetra! Et...   
1101  Interdum lacus etiam tellus viverra? Molestie....   

                                                    Q16  \
2     Accusamus 

In [105]:
df.shape

(1102, 38)

In [106]:
list(df_clean)

['StartDate',
 'EndDate',
 'Status',
 'IPAddress',
 'Progress',
 'Duration (in seconds)',
 'Finished',
 'RecordedDate',
 'ResponseId',
 'RecipientLastName',
 'RecipientFirstName',
 'RecipientEmail',
 'ExternalReference',
 'LocationLatitude',
 'LocationLongitude',
 'DistributionChannel',
 'UserLanguage',
 'Q1',
 'Q2',
 'Q4',
 'Q5',
 'Q6',
 'Q7',
 'Q8',
 'Q9',
 'Q10',
 'Q11',
 'Q12',
 'Q13_1',
 'Q13_2',
 'Q13_3',
 'Q13_4',
 'Q14',
 'Q15',
 'Q16',
 'Q17',
 'Q18',
 'Q19']

In [107]:
df_clean.columns

Index(['StartDate', 'EndDate', 'Status', 'IPAddress', 'Progress',
       'Duration (in seconds)', 'Finished', 'RecordedDate', 'ResponseId',
       'RecipientLastName', 'RecipientFirstName', 'RecipientEmail',
       'ExternalReference', 'LocationLatitude', 'LocationLongitude',
       'DistributionChannel', 'UserLanguage', 'Q1', 'Q2', 'Q4', 'Q5', 'Q6',
       'Q7', 'Q8', 'Q9', 'Q10', 'Q11', 'Q12', 'Q13_1', 'Q13_2', 'Q13_3',
       'Q13_4', 'Q14', 'Q15', 'Q16', 'Q17', 'Q18', 'Q19'],
      dtype='object')

In [108]:
df_metadata = df_clean[['ResponseId','RecordedDate']]
df_metadata

ResponseId         RecordedDate
2     R_6fn5ekDobUP1m5g  2025-07-31 05:41:54
3     R_3KR0mBnSZdjoJAa  2025-07-31 05:41:55
4     R_5gqVnk14OjcBmXI  2025-07-31 05:41:55
5     R_0HUEeMmeNFs5WZw  2025-07-31 05:41:55
6     R_7TYgOzdHL53Y5Ho  2025-07-31 05:41:55
...                 ...                  ...
1097  R_8x0fqyHU55k51Eq  2025-07-31 17:04:12
1098  R_72OeMk6x3Bgz7AG  2025-07-31 17:04:13
1099  R_2rzg4pqMXolgfbg  2025-07-31 17:04:13
1100  R_6G9pvj6waxWi4RM  2025-07-31 17:04:13
1101  R_9GEi9QDgBzaApEi  2025-07-31 17:04:13

[1100 rows x 2 columns]

In [109]:
df_simple= df_clean[['ResponseId', 'Q1',
 'Q2',
 'Q4',
 'Q5',
 'Q6',
 'Q7',
 'Q8',
 'Q9',
 'Q10',
 'Q11',
 'Q12',
 'Q13_1',
 'Q13_2',
 'Q13_3',
 'Q13_4',]]
df_simple

ResponseId Q1 Q2 Q4 Q5 Q6 Q7 Q8 Q9 Q10 Q11 Q12 Q13_1 Q13_2 Q13_3  \
2     R_6fn5ekDobUP1m5g  4  3  1  2  2  3  3  3   2   2   4     1     4     5   
3     R_3KR0mBnSZdjoJAa  1  1  5  3  2  1  3  3   1   2   4     3     2     1   
4     R_5gqVnk14OjcBmXI  4  3  4  4  2  3  3  3   2   1   4     4     3     3   
5     R_0HUEeMmeNFs5WZw  3  3  2  2  4  1  1  1   3   4   5     1     2     5   
6     R_7TYgOzdHL53Y5Ho  1  3  1  3  2  1  4  1   2   3   4     3     4     3   
...                 ... .. .. .. .. .. .. .. ..  ..  ..  ..   ...   ...   ...   
1097  R_8x0fqyHU55k51Eq  4  1  2  3  1  2  5  3   3   2   5     3     3     1   
1098  R_72OeMk6x3Bgz7AG  2  3  4  5  2  2  1  1   2   4   2     4     3     1   
1099  R_2rzg4pqMXolgfbg  1  1  4  2  1  1  5  1   3   3   2     3     2     3   
1100  R_6G9pvj6waxWi4RM  3  2  5  1  5  3  3  1   1   2   2     5     2     4   
1101  R_9GEi9QDgBzaApEi  1  2  1  5  1  1  1  1   2   3   2     3     2     4   

     Q13_4  
2        3  
3        1  
4        4  
5        4  
6        5  
...    ...  
1097     1  
1098     2  
1099     2  
1100     4  
1101     5  

[1100 rows x 16 columns]

In [110]:
df_simple['Q1'].describe()

count     1100
unique       5
top          5
freq       243
Name: Q1, dtype: object

tells us theres 5 unique values, the top value is 5, the frequency of values we have is 243

In [111]:
df_simple= df_simple.set_index("ResponseId")
df_simple

Q1 Q2 Q4 Q5 Q6 Q7 Q8 Q9 Q10 Q11 Q12 Q13_1 Q13_2 Q13_3 Q13_4
ResponseId                                                                   
R_6fn5ekDobUP1m5g  4  3  1  2  2  3  3  3   2   2   4     1     4     5     3
R_3KR0mBnSZdjoJAa  1  1  5  3  2  1  3  3   1   2   4     3     2     1     1
R_5gqVnk14OjcBmXI  4  3  4  4  2  3  3  3   2   1   4     4     3     3     4
R_0HUEeMmeNFs5WZw  3  3  2  2  4  1  1  1   3   4   5     1     2     5     4
R_7TYgOzdHL53Y5Ho  1  3  1  3  2  1  4  1   2   3   4     3     4     3     5
...               .. .. .. .. .. .. .. ..  ..  ..  ..   ...   ...   ...   ...
R_8x0fqyHU55k51Eq  4  1  2  3  1  2  5  3   3   2   5     3     3     1     1
R_72OeMk6x3Bgz7AG  2  3  4  5  2  2  1  1   2   4   2     4     3     1     2
R_2rzg4pqMXolgfbg  1  1  4  2  1  1  5  1   3   3   2     3     2     3     2
R_6G9pvj6waxWi4RM  3  2  5  1  5  3  3  1   1   2   2     5     2     4     4
R_9GEi9QDgBzaApEi  1  2  1  5  1  1  1  1   2   3   2     3     2     4     5

[1100 rows x 15 columns]

In [112]:
df_simple.describe()

Q1    Q2    Q4    Q5    Q6    Q7    Q8    Q9   Q10   Q11   Q12  \
count   1100  1100  1100  1100  1100  1100  1100  1100  1100  1100  1100   
unique     5     3     5     5     5     3     5     3     3     5     5   
top        5     3     2     5     2     2     4     3     2     2     4   
freq     243   385   238   253   237   388   256   371   405   236   240   

       Q13_1 Q13_2 Q13_3 Q13_4  
count   1100  1100  1100  1100  
unique     5     5     5     5  
top        3     3     5     4  
freq     247   231   240   242

In [113]:
df_simple =df_simple.astype(int)
df_simple

Q1  Q2  Q4  Q5  Q6  Q7  Q8  Q9  Q10  Q11  Q12  Q13_1  \
ResponseId                                                                
R_6fn5ekDobUP1m5g   4   3   1   2   2   3   3   3    2    2    4      1   
R_3KR0mBnSZdjoJAa   1   1   5   3   2   1   3   3    1    2    4      3   
R_5gqVnk14OjcBmXI   4   3   4   4   2   3   3   3    2    1    4      4   
R_0HUEeMmeNFs5WZw   3   3   2   2   4   1   1   1    3    4    5      1   
R_7TYgOzdHL53Y5Ho   1   3   1   3   2   1   4   1    2    3    4      3   
...                ..  ..  ..  ..  ..  ..  ..  ..  ...  ...  ...    ...   
R_8x0fqyHU55k51Eq   4   1   2   3   1   2   5   3    3    2    5      3   
R_72OeMk6x3Bgz7AG   2   3   4   5   2   2   1   1    2    4    2      4   
R_2rzg4pqMXolgfbg   1   1   4   2   1   1   5   1    3    3    2      3   
R_6G9pvj6waxWi4RM   3   2   5   1   5   3   3   1    1    2    2      5   
R_9GEi9QDgBzaApEi   1   2   1   5   1   1   1   1    2    3    2      3   

                   Q13_2  Q13_3  Q13_4  
ResponseId                              
R_6fn5ekDobUP1m5g      4      5      3  
R_3KR0mBnSZdjoJAa      2      1      1  
R_5gqVnk14OjcBmXI      3      3      4  
R_0HUEeMmeNFs5WZw      2      5      4  
R_7TYgOzdHL53Y5Ho      4      3      5  
...                  ...    ...    ...  
R_8x0fqyHU55k51Eq      3      1      1  
R_72OeMk6x3Bgz7AG      3      1      2  
R_2rzg4pqMXolgfbg      2      3      2  
R_6G9pvj6waxWi4RM      2      4      4  
R_9GEi9QDgBzaApEi      2      4      5  

[1100 rows x 15 columns]

In [114]:
### recoding for questions 7, 9, 10

recode_dict_v1 = {
    1:3,
    2:1,
    3:2
}
recode_dict_v1

{1: 3, 2: 1, 3: 2}

In [115]:
## use recode_dict_v1 for Q7, 9, 10

df_simple_recode= df_simple.copy()
df_simple_recode["Q7"] = df_simple_recode["Q7"].map(recode_dict_v1)
df_simple_recode["Q9"] = df_simple_recode["Q9"].map(recode_dict_v1)
df_simple_recode["Q10"] = df_simple_recode["Q10"].map(recode_dict_v1)

df_simple_recode

Q1  Q2  Q4  Q5  Q6  Q7  Q8  Q9  Q10  Q11  Q12  Q13_1  \
ResponseId                                                                
R_6fn5ekDobUP1m5g   4   3   1   2   2   2   3   2    1    2    4      1   
R_3KR0mBnSZdjoJAa   1   1   5   3   2   3   3   2    3    2    4      3   
R_5gqVnk14OjcBmXI   4   3   4   4   2   2   3   2    1    1    4      4   
R_0HUEeMmeNFs5WZw   3   3   2   2   4   3   1   3    2    4    5      1   
R_7TYgOzdHL53Y5Ho   1   3   1   3   2   3   4   3    1    3    4      3   
...                ..  ..  ..  ..  ..  ..  ..  ..  ...  ...  ...    ...   
R_8x0fqyHU55k51Eq   4   1   2   3   1   1   5   2    2    2    5      3   
R_72OeMk6x3Bgz7AG   2   3   4   5   2   1   1   3    1    4    2      4   
R_2rzg4pqMXolgfbg   1   1   4   2   1   3   5   3    2    3    2      3   
R_6G9pvj6waxWi4RM   3   2   5   1   5   2   3   3    3    2    2      5   
R_9GEi9QDgBzaApEi   1   2   1   5   1   3   1   3    1    3    2      3   

                   Q13_2  Q13_3  Q13_4  
ResponseId                              
R_6fn5ekDobUP1m5g      4      5      3  
R_3KR0mBnSZdjoJAa      2      1      1  
R_5gqVnk14OjcBmXI      3      3      4  
R_0HUEeMmeNFs5WZw      2      5      4  
R_7TYgOzdHL53Y5Ho      4      3      5  
...                  ...    ...    ...  
R_8x0fqyHU55k51Eq      3      1      1  
R_72OeMk6x3Bgz7AG      3      1      2  
R_2rzg4pqMXolgfbg      2      3      2  
R_6G9pvj6waxWi4RM      2      4      4  
R_9GEi9QDgBzaApEi      2      4      5  

[1100 rows x 15 columns]

In [116]:
### recoding for question 11
recode_dict_v2= {
    1:1,
    4:1,
    5:1,
    2:2,
    3:2
}
recode_dict_v2

{1: 1, 4: 1, 5: 1, 2: 2, 3: 2}

In [117]:
##use recode_dict_v1 for Q11

df_simple_recode_v2= df_simple_recode.copy()
df_simple_recode_v2["Q11"] = df_simple_recode_v2["Q11"].map(recode_dict_v2)

df_simple_recode

Q1  Q2  Q4  Q5  Q6  Q7  Q8  Q9  Q10  Q11  Q12  Q13_1  \
ResponseId                                                                
R_6fn5ekDobUP1m5g   4   3   1   2   2   2   3   2    1    2    4      1   
R_3KR0mBnSZdjoJAa   1   1   5   3   2   3   3   2    3    2    4      3   
R_5gqVnk14OjcBmXI   4   3   4   4   2   2   3   2    1    1    4      4   
R_0HUEeMmeNFs5WZw   3   3   2   2   4   3   1   3    2    4    5      1   
R_7TYgOzdHL53Y5Ho   1   3   1   3   2   3   4   3    1    3    4      3   
...                ..  ..  ..  ..  ..  ..  ..  ..  ...  ...  ...    ...   
R_8x0fqyHU55k51Eq   4   1   2   3   1   1   5   2    2    2    5      3   
R_72OeMk6x3Bgz7AG   2   3   4   5   2   1   1   3    1    4    2      4   
R_2rzg4pqMXolgfbg   1   1   4   2   1   3   5   3    2    3    2      3   
R_6G9pvj6waxWi4RM   3   2   5   1   5   2   3   3    3    2    2      5   
R_9GEi9QDgBzaApEi   1   2   1   5   1   3   1   3    1    3    2      3   

                   Q13_2  Q13_3  Q13_4  
ResponseId                              
R_6fn5ekDobUP1m5g      4      5      3  
R_3KR0mBnSZdjoJAa      2      1      1  
R_5gqVnk14OjcBmXI      3      3      4  
R_0HUEeMmeNFs5WZw      2      5      4  
R_7TYgOzdHL53Y5Ho      4      3      5  
...                  ...    ...    ...  
R_8x0fqyHU55k51Eq      3      1      1  
R_72OeMk6x3Bgz7AG      3      1      2  
R_2rzg4pqMXolgfbg      2      3      2  
R_6G9pvj6waxWi4RM      2      4      4  
R_9GEi9QDgBzaApEi      2      4      5  

[1100 rows x 15 columns]

In [118]:
df_simple_recode['total_score'] = df_simple_recode.sum(axis=1)
df_simple_recode

Q1  Q2  Q4  Q5  Q6  Q7  Q8  Q9  Q10  Q11  Q12  Q13_1  \
ResponseId                                                                
R_6fn5ekDobUP1m5g   4   3   1   2   2   2   3   2    1    2    4      1   
R_3KR0mBnSZdjoJAa   1   1   5   3   2   3   3   2    3    2    4      3   
R_5gqVnk14OjcBmXI   4   3   4   4   2   2   3   2    1    1    4      4   
R_0HUEeMmeNFs5WZw   3   3   2   2   4   3   1   3    2    4    5      1   
R_7TYgOzdHL53Y5Ho   1   3   1   3   2   3   4   3    1    3    4      3   
...                ..  ..  ..  ..  ..  ..  ..  ..  ...  ...  ...    ...   
R_8x0fqyHU55k51Eq   4   1   2   3   1   1   5   2    2    2    5      3   
R_72OeMk6x3Bgz7AG   2   3   4   5   2   1   1   3    1    4    2      4   
R_2rzg4pqMXolgfbg   1   1   4   2   1   3   5   3    2    3    2      3   
R_6G9pvj6waxWi4RM   3   2   5   1   5   2   3   3    3    2    2      5   
R_9GEi9QDgBzaApEi   1   2   1   5   1   3   1   3    1    3    2      3   

                   Q13_2  Q13_3  Q13_4  total_score  
ResponseId                                           
R_6fn5ekDobUP1m5g      4      5      3           39  
R_3KR0mBnSZdjoJAa      2      1      1           36  
R_5gqVnk14OjcBmXI      3      3      4           44  
R_0HUEeMmeNFs5WZw      2      5      4           44  
R_7TYgOzdHL53Y5Ho      4      3      5           43  
...                  ...    ...    ...          ...  
R_8x0fqyHU55k51Eq      3      1      1           36  
R_72OeMk6x3Bgz7AG      3      1      2           38  
R_2rzg4pqMXolgfbg      2      3      2           37  
R_6G9pvj6waxWi4RM      2      4      4           46  
R_9GEi9QDgBzaApEi      2      4      5           37  

[1100 rows x 16 columns]

In [119]:
# q2, 4, 5, 12, 13 summation for enviornment_satisfcation

df_simple_recode['environment_satisfaction'] = df_simple_recode['Q2'] + df_simple_recode['Q4'] + df_simple_recode['Q5'] + df_simple_recode['Q12'] + df_simple_recode['Q13_1'] + df_simple_recode['Q13_2'] + df_simple_recode['Q13_3'] + df_simple_recode['Q13_4']

df_simple_recode

Q1  Q2  Q4  Q5  Q6  Q7  Q8  Q9  Q10  Q11  Q12  Q13_1  \
ResponseId                                                                
R_6fn5ekDobUP1m5g   4   3   1   2   2   2   3   2    1    2    4      1   
R_3KR0mBnSZdjoJAa   1   1   5   3   2   3   3   2    3    2    4      3   
R_5gqVnk14OjcBmXI   4   3   4   4   2   2   3   2    1    1    4      4   
R_0HUEeMmeNFs5WZw   3   3   2   2   4   3   1   3    2    4    5      1   
R_7TYgOzdHL53Y5Ho   1   3   1   3   2   3   4   3    1    3    4      3   
...                ..  ..  ..  ..  ..  ..  ..  ..  ...  ...  ...    ...   
R_8x0fqyHU55k51Eq   4   1   2   3   1   1   5   2    2    2    5      3   
R_72OeMk6x3Bgz7AG   2   3   4   5   2   1   1   3    1    4    2      4   
R_2rzg4pqMXolgfbg   1   1   4   2   1   3   5   3    2    3    2      3   
R_6G9pvj6waxWi4RM   3   2   5   1   5   2   3   3    3    2    2      5   
R_9GEi9QDgBzaApEi   1   2   1   5   1   3   1   3    1    3    2      3   

                   Q13_2  Q13_3  Q13_4  total_score  environment_satisfaction  
ResponseId                                                                     
R_6fn5ekDobUP1m5g      4      5      3           39                        23  
R_3KR0mBnSZdjoJAa      2      1      1           36                        20  
R_5gqVnk14OjcBmXI      3      3      4           44                        29  
R_0HUEeMmeNFs5WZw      2      5      4           44                        24  
R_7TYgOzdHL53Y5Ho      4      3      5           43                        26  
...                  ...    ...    ...          ...                       ...  
R_8x0fqyHU55k51Eq      3      1      1           36                        19  
R_72OeMk6x3Bgz7AG      3      1      2           38                        24  
R_2rzg4pqMXolgfbg      2      3      2           37                        19  
R_6G9pvj6waxWi4RM      2      4      4           46                        25  
R_9GEi9QDgBzaApEi      2      4      5           37                        24  

[1100 rows x 17 columns]

In [120]:
# q8 and q10 for routine_satisfaction

df_simple_recode['routine_satisfaction'] = df_simple_recode['Q8'] + df_simple_recode['Q10']

df_simple_recode

Q1  Q2  Q4  Q5  Q6  Q7  Q8  Q9  Q10  Q11  Q12  Q13_1  \
ResponseId                                                                
R_6fn5ekDobUP1m5g   4   3   1   2   2   2   3   2    1    2    4      1   
R_3KR0mBnSZdjoJAa   1   1   5   3   2   3   3   2    3    2    4      3   
R_5gqVnk14OjcBmXI   4   3   4   4   2   2   3   2    1    1    4      4   
R_0HUEeMmeNFs5WZw   3   3   2   2   4   3   1   3    2    4    5      1   
R_7TYgOzdHL53Y5Ho   1   3   1   3   2   3   4   3    1    3    4      3   
...                ..  ..  ..  ..  ..  ..  ..  ..  ...  ...  ...    ...   
R_8x0fqyHU55k51Eq   4   1   2   3   1   1   5   2    2    2    5      3   
R_72OeMk6x3Bgz7AG   2   3   4   5   2   1   1   3    1    4    2      4   
R_2rzg4pqMXolgfbg   1   1   4   2   1   3   5   3    2    3    2      3   
R_6G9pvj6waxWi4RM   3   2   5   1   5   2   3   3    3    2    2      5   
R_9GEi9QDgBzaApEi   1   2   1   5   1   3   1   3    1    3    2      3   

                   Q13_2  Q13_3  Q13_4  total_score  environment_satisfaction  \
ResponseId                                                                      
R_6fn5ekDobUP1m5g      4      5      3           39                        23   
R_3KR0mBnSZdjoJAa      2      1      1           36                        20   
R_5gqVnk14OjcBmXI      3      3      4           44                        29   
R_0HUEeMmeNFs5WZw      2      5      4           44                        24   
R_7TYgOzdHL53Y5Ho      4      3      5           43                        26   
...                  ...    ...    ...          ...                       ...   
R_8x0fqyHU55k51Eq      3      1      1           36                        19   
R_72OeMk6x3Bgz7AG      3      1      2           38                        24   
R_2rzg4pqMXolgfbg      2      3      2           37                        19   
R_6G9pvj6waxWi4RM      2      4      4           46                        25   
R_9GEi9QDgBzaApEi      2      4      5           37                        24   

                   routine_satisfaction  
ResponseId                               
R_6fn5ekDobUP1m5g                     4  
R_3KR0mBnSZdjoJAa                     6  
R_5gqVnk14OjcBmXI                     4  
R_0HUEeMmeNFs5WZw                     3  
R_7TYgOzdHL53Y5Ho                     5  
...                                 ...  
R_8x0fqyHU55k51Eq                     7  
R_72OeMk6x3Bgz7AG                     2  
R_2rzg4pqMXolgfbg                     7  
R_6G9pvj6waxWi4RM                     6  
R_9GEi9QDgBzaApEi                     2  

[1100 rows x 18 columns]

In [122]:
# 1, 2, 6, 9, 10 for training_quality_satisfaction

df_simple_recode['training_quality_satisfaction'] = df_simple_recode['Q1'] + df_simple_recode['Q2'] + df_simple_recode['Q6'] + df_simple_recode['Q9'] + df_simple_recode['Q10']

df_simple_recode

Q1  Q2  Q4  Q5  Q6  Q7  Q8  Q9  Q10  Q11  Q12  Q13_1  \
ResponseId                                                                
R_6fn5ekDobUP1m5g   4   3   1   2   2   2   3   2    1    2    4      1   
R_3KR0mBnSZdjoJAa   1   1   5   3   2   3   3   2    3    2    4      3   
R_5gqVnk14OjcBmXI   4   3   4   4   2   2   3   2    1    1    4      4   
R_0HUEeMmeNFs5WZw   3   3   2   2   4   3   1   3    2    4    5      1   
R_7TYgOzdHL53Y5Ho   1   3   1   3   2   3   4   3    1    3    4      3   
...                ..  ..  ..  ..  ..  ..  ..  ..  ...  ...  ...    ...   
R_8x0fqyHU55k51Eq   4   1   2   3   1   1   5   2    2    2    5      3   
R_72OeMk6x3Bgz7AG   2   3   4   5   2   1   1   3    1    4    2      4   
R_2rzg4pqMXolgfbg   1   1   4   2   1   3   5   3    2    3    2      3   
R_6G9pvj6waxWi4RM   3   2   5   1   5   2   3   3    3    2    2      5   
R_9GEi9QDgBzaApEi   1   2   1   5   1   3   1   3    1    3    2      3   

                   Q13_2  Q13_3  Q13_4  total_score  environment_satisfaction  \
ResponseId                                                                      
R_6fn5ekDobUP1m5g      4      5      3           39                        23   
R_3KR0mBnSZdjoJAa      2      1      1           36                        20   
R_5gqVnk14OjcBmXI      3      3      4           44                        29   
R_0HUEeMmeNFs5WZw      2      5      4           44                        24   
R_7TYgOzdHL53Y5Ho      4      3      5           43                        26   
...                  ...    ...    ...          ...                       ...   
R_8x0fqyHU55k51Eq      3      1      1           36                        19   
R_72OeMk6x3Bgz7AG      3      1      2           38                        24   
R_2rzg4pqMXolgfbg      2      3      2           37                        19   
R_6G9pvj6waxWi4RM      2      4      4           46                        25   
R_9GEi9QDgBzaApEi      2      4      5           37                        24   

                   routine_satisfaction  training_quality_satisfaction  
ResponseId                                                              
R_6fn5ekDobUP1m5g                     4                             12  
R_3KR0mBnSZdjoJAa                     6                              9  
R_5gqVnk14OjcBmXI                     4                             12  
R_0HUEeMmeNFs5WZw                     3                             15  
R_7TYgOzdHL53Y5Ho                     5                             10  
...                                 ...                            ...  
R_8x0fqyHU55k51Eq                     7                             10  
R_72OeMk6x3Bgz7AG                     2                             11  
R_2rzg4pqMXolgfbg                     7                              8  
R_6G9pvj6waxWi4RM                     6                             16  
R_9GEi9QDgBzaApEi                     2                              8  

[1100 rows x 19 columns]

In [123]:
df_simple_scores = df_simple_recode[['total_score', 'environment_satisfaction', 'routine_satisfaction', 'training_quality_satisfaction']]

df_simple_scores

total_score  environment_satisfaction  \
ResponseId                                                 
R_6fn5ekDobUP1m5g           39                        23   
R_3KR0mBnSZdjoJAa           36                        20   
R_5gqVnk14OjcBmXI           44                        29   
R_0HUEeMmeNFs5WZw           44                        24   
R_7TYgOzdHL53Y5Ho           43                        26   
...                        ...                       ...   
R_8x0fqyHU55k51Eq           36                        19   
R_72OeMk6x3Bgz7AG           38                        24   
R_2rzg4pqMXolgfbg           37                        19   
R_6G9pvj6waxWi4RM           46                        25   
R_9GEi9QDgBzaApEi           37                        24   

                   routine_satisfaction  training_quality_satisfaction  
ResponseId                                                              
R_6fn5ekDobUP1m5g                     4                             12  
R_3KR0mBnSZdjoJAa                     6                              9  
R_5gqVnk14OjcBmXI                     4                             12  
R_0HUEeMmeNFs5WZw                     3                             15  
R_7TYgOzdHL53Y5Ho                     5                             10  
...                                 ...                            ...  
R_8x0fqyHU55k51Eq                     7                             10  
R_72OeMk6x3Bgz7AG                     2                             11  
R_2rzg4pqMXolgfbg                     7                              8  
R_6G9pvj6waxWi4RM                     6                             16  
R_9GEi9QDgBzaApEi                     2                              8  

[1100 rows x 4 columns]

In [ ]:
## save to csv
df_simple.to_csv("simple.csv")